### PR TITLE
2wp improvements

### PR DESCRIPTION
**Pull request by josedahlquist**
This PR implements improvements to the 2 way peg.

Adds support to locking from p2sh-p2wpkh bitcoin addresses! (segwit compatible mode)
Allows refunding to multisig bitcoin addresses, both legacy and segwit.
Adds new methods to Bridge:
- registerBtcCoinbaseTransaction: Registers crucial information for an already registered block's coinbase transaction. This information is then used when registering a segwit transaction.
- hasBtcBlockCoinbaseTransactionInformation: informs if the coinbase for a given block was informed or not.
Fixes problems with performance tests, and patches other problems.

*fed: 2wp-segwit*
*fit: 2wp-segwit*

### CODE REVIEW

**Comment by josedahlquist** [comment]
pipeline:run

**Comment by josedahlquist** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java [hunk 1]
```diff
@@ -707,6 +708,30 @@ public static Integer deserializeInteger(byte[] data) {
...
 
+    public static CoinbaseInformation deserializeCoinbaseInformation(byte[] data) {
+        if (data == null) {
+            return null;
+        }
+        RLPList rlpList = (RLPList)RLP.decode2(data).get(0);
+
+        if (rlpList.size() != 1) {
+            throw new RuntimeException(String.format("Invalid serialized coinbase information, expected 2 values but got %n", rlpList.size()));
+        }
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java]: Throws exception if size != 1 but the message says it expects size to be 2

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 2]
```diff
@@ -324,21 +337,26 @@ public void registerBtcTransaction(Transaction rskTx, byte[] btcTxSerialized, in
         BtcTransaction btcTx = new BtcTransaction(bridgeConstants.getBtcParams(), btcTxSerialized);
         btcTx.verify();
 
+        // Check the again tx was not already processed but making sure to use the txid (no witness)
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Confusing comment

**Comment by josedahlquist** [comment]
pipeline:run

**Comment by josedahlquist** [comment]
pipeline:run

**Comment by josedahlquist** [comment]
pipeline:run

**Pull request merged**